PlaywrigthUITests/Homework/SmartWait.cs

```diff
@@ -17,9 +17,7 @@ public async Task Nike()
             await Page.GetByRole(AriaRole.Button, new() { Name = "Color", Exact = true }).ClickAsync();
             var titelsBefore = await Page.Locator(".product-card__title").AllInnerTextsAsync();
             await Page.GetByLabel("Filter for Blue").ClickAsync();
-            
-            await Page.WaitForTimeoutAsync(5000); // Replace this line with smart waiting
-
+            await Page.WaitForSelectorAsync("//h1[contains(text(), 'Womens Blue Shoes')]");
```

Comment:
At first, I did it by ID - but I remembered your words that practice shows that "text is more reliable" + checking the header of a page at once.

Reply:
"text is more reliable" means that if you have button "Submit" with class='submit' then it looks more reliable to use text for this particular element because more likely that class would change rather than text. If we talk about "some" element, like some product which appears after filtering then much more likely that today you see one product and tomorrow another product (not that product name itself change, but that another product will be on the page) - in this case it more reliable to use some product placeholder (some elements which is the same for different products)